rfc/004-feature-template.md

```diff
@@ -0,0 +1,54 @@
+- Feature Name: badges_node
+- Start Date: 2020-08-24
+- RFC PR: [amundsen-io/rfcs#4](https://github.com/amundsen-io/rfcs/pull/4)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000)
+
+# Separate badges from tags
+
+## Summary
+
+Create a new node type for neo4j which would represent Badges that is separate from the Tag node.
+
+## Motivation
+
+In order to best use badges we want to add sentiment and category fields to the Badge nodes. This makes the Tag and Badge components different, as Badge requires more information now. Separating the two now will likely lead to less tech debt in the future when other features are implemented that impact the way we ingest and use badges. Additionally, this change will help clarify the difference in use and purpose between tags and badges throughtout Amundsen. This feature will also guarantee badges are not ingested without having sentiment and category.
+
+
+## Guide-level Explanation (aka Product Details)
+
+> Explain the proposal as if it was already included in Amundsen and you were teaching it to an Amundsen user. That generally means:
+
+> Introducing new named concepts.
+> Explaining the feature largely in terms of examples.
+> Explaining how Amundsen users should think about the feature, and how it should impact the way they use Amundsen. It should explain the impact as concretely as possible.
+> If applicable, provide deprecation warnings, or migration guidance.
+> For implementation-oriented RFCs, this section should focus on how maintainers should think about the change, and give examples of its concrete impact. For policy RFCs, this section should provide an example-driven introduction to the policy, and explain its impact in concrete terms.
+- In the future this feature will hopefully replace all use of old Tag badges.
+
+## Reference-level Explanation (aka Technical Details)
+
+The implementation will require:
+- Add methods to the metadata service API to add, remove and, get badges as currently the service uses the methods for tags.
+- Update badge ingestion docs to reflect new API call and parameters.
+- Make updates to search, databuilder and frontend service to add functionality to use Badge instead of Tag.
```

Comment:
I think we could expand the details here with a little bit more context : how search is handling badge and what is the new proposed , same for fe, databuilder, etc?

We could have the implementation prs going in the mean time while keep updating these section while ready. WDYT?

Reply:
I think that's a good idea since I don't yet have enough context about how search and databuilder, I will add more detail about FE now.

Reply:
yeah, that's why I think once you touch all the places(FE, search, metadata, databuilder) where badge is used, you could update this doc correspondingly. In the end, I think RFC should be used to document the context of why, where, and how we propose the change.